poc/vdaf_poc/field.py

```diff
@@ -166,6 +168,79 @@ class NttField(Field):
     def gen(cls) -> Self:
         raise NotImplementedError()
 
+    @classmethod
+    def nth_root(cls, n: int) -> Self:
+        """Returns the $2^n$-th root of unity."""
+        return cls.gen() ** (cls.GEN_ORDER >> n)
+
+    @classmethod
+    def nth_root_powers(cls, n: int) -> list[Self]:
+        """Returns the powers of the $n$-th root of unity."""
+        log_n = assert_power_of_2(n)
+        root = cls.nth_root(log_n)
+        return [root**i for i in range(n)]
+
+    @classmethod
+    def ntt(cls, p: list[Self], n: int) -> list[Self]:
+        r"""
+        Number Theoretic Transform (NTT) over a field.
+
+        Given a polynomial P in monomial basis, returns
+        $\\{ P({w_n}^i) : 0 \\leq i < n \\}$
+        where $w_n$ is the $n$-th root of unity.
```

Comment:
We're going to end up copy-pasting this into the draft. Let's try to make this readable without the latex formatting.